### PR TITLE
Cards, Video and Audio components

### PR DESCRIPTION
**Pull request by olemartinorg**
## Description

See #2024 for the original PR. This includes minor adjustments and tests.

Noteworthy changes from #2024:
- Instead of a `Card` and `CardGroup`, these were combined into `Cards` that can have multiple cards configured at once (it didn't make that much sense to optimize for rendering one Card alone, and making a grouping of cards an extra configuration).
- Instead of having options to configure Image/Audio/Video as media in each card, separate `Audio` and `Video` components were created (that can also stand on their own and render outside cards). We already have an `Image` component that could be used for card media.
- Cards can include other non-repeating components (pretty much like a `Group`, but most similar to `Grid`). These will be displayed under the title and/or description.
- Some QoL adjustments and fixes:
  - Added default values for some configuration options, hopefully making Cards look good without too much configuration and confusion (i.e. media height was 0 by default, making them disappear - which was a bit confusing).
  - Renamed `textResourceBindings` in `Audio` and `Video` from `altTextAudio` and `altTextVideo` to just `altText`.
  - Removed the `wwwroot` trick that made it seem like you could reference URLs releative to the root of the app (but URLs can already be relative, and the point of having a root is that you don't have to include its name).
  - Removed the helpText-support in `Audio` and `Video`. I think these should probably be rendered via a label instead (in case we'll render out labels for these in the future), and removing it simplified the code significantly and removed usages of the mui `Grid`.
- Added a Cypress-test, added the component to frontend-test (both outside and inside repeating groups), and taking a snapshot of the Cards components so we'll be notified of any visual regressions.

## Related Issue(s)

- closes #{issue number}

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [x] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [x] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [ ] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [ ] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [ ] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

**Comment by bjosttveit** [comment]
>     * double-signing failed, but I've seen that test failing in main as well.

Yep that is due to https://github.com/Altinn/app-frontend-react/issues/2091 which gets fixed in https://github.com/Altinn/app-frontend-react/pull/2108

File: src/layout/Cards/CardContext.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  minMediaHeight?: string;
+}
+
+const { Provider, useCtx } = createContext<CardContext | undefined>({
+  name: 'Card',
+  required: false,
+  default: undefined,
+});
+
+export function CardProvider({ children, ...state }: PropsWithChildren<CardContext>) {
```
Comment (cammiida) [inline src/layout/Cards/CardContext.tsx:19]: Meg og min masing om contexter igjen 🙈 
Går det an å heller bruke en funksjon ala dette i stedet for contexten som `Video`, `Audio` og `ImageComponent` bruker? Da kan man lettere endre på hver enkelt om behovene endres 😊 
```typescript
export function getCardsConfig(node: LayoutNode, defaultHeight = '150px') {
  const cardsParent = findParentNodeOfType<LayoutNode<'Cards'>>(node, 'Cards');
  const cardMediaHeight = parseSize(cardsParent?.item.minMediaHeight, defaultHeight);

  return { hasCardsParent: !!cardsParent, cardMediaHeight };
}
```

**Review by cammiida — APPROVED** [review]
Noen spørsmål om potensielle endringer. Ellers ser det veldig bra ut for meg 🎸 ✨

File: src/codegen/ComponentConfig.ts [hunk 2]
```diff
@@ -24,6 +24,8 @@ export interface RequiredComponentConfig {
     renderInButtonGroup: boolean;
     renderInAccordion: boolean;
     renderInAccordionGroup: boolean;
+    renderInCards: boolean;
+    renderInCardsMedia: boolean;
```
[the remark this answers is not in the document]
Comment (olemartinorg) [reply src/codegen/ComponentConfig.ts:28]: Ah, ja, det er jeg enig i. Samtidig er det litt vanskelig å vite hva som skal være default-verdi. For `renderInButtonGroup` burde jo defaulten være `false` (det meste av nye komponenter er noe annet enn buttons?), og forsåvidt det samme for `renderInCardsMedia`, men for f.eks. `renderInTable` tenker jeg det kanskje passer bedre med `true`.. :thinking: Fordelen med å tvinge det er at man må tenke over hvordan hver komponent må forholde seg til hver av de andre komponentene, og hvis det ikke er påkrevd er det mye lettere å bare glemme å sette verdiene hvis man lager noe nytt (og da gir jo egentlig ikke verdiene like mye verdi lengre).

**Pull request merged**